test/functional/tools/inputs_as_json.xml

```diff
@@ -0,0 +1,79 @@
+<tool id="inputs_as_json" name="inputs_as_json" version="1.0.0">
+    <configfiles>
+        <inputs name="inputs" format="json" version="1" />
+        <configfile name="check_inputs"><![CDATA[
+import json
+import sys
+
+input_json_path = sys.argv[1]
+test_case = sys.argv[2]
+as_dict = json.load(open(input_json_path, "r"))
+
+
+def assert_equals(x, y):
+    assert x == y, "%s != %s" % (x, y)
+
+if test_case == "1":
+    assert_equals(as_dict["test_case"], 1)
+    assert_equals(as_dict["text_test"], "foo")
+    assert_equals(as_dict["booltest"], True)
+    assert_equals(as_dict["inttest"], 12456)
+    assert_equals(as_dict["floattest"], 6.789)
+    assert_equals(as_dict["radio_select"], "a_radio")
+elif test_case == "2":
+    assert_equals(as_dict["test_case"], 2)
+    assert_equals(as_dict["text_test"], "bar")
+    assert_equals(as_dict["booltest"], False)
+    assert_equals(as_dict["inttest"], None)
+    assert_equals(as_dict["floattest"], 1.0)
+    assert_equals(as_dict["radio_select"], "a_radio")
+
+with open("output", "w") as f:
+    f.write("okay\n")
+]]></configfile>
+    </configfiles>
+    <command>
+        python $check_inputs $inputs $test_case
+    </command>
+    <inputs>
+        <param name="test_case" type="integer" value="0" />
+        <param name="text_test" type="text" />
+        <param name="booltest" truevalue="booltrue" falsevalue="boolfalse" checked="false" type="boolean" />
+        <param name="inttest" type="integer" optional="true" />
```

Comment:
For documentation purpose can we include a conditional here?
Especially in the case of a conditional the we don't want to include a large list of `None`s, isn't it?